domain/secretbackend/watcher_test.go

```diff
@@ -150,10 +150,13 @@ func (s *watcherSuite) TestWatchSecretBackendRotationChanges(c *gc.C) {
 	err = state.DeleteSecretBackend(context.Background(), secretbackend.BackendIdentifier{ID: backendID2}, false)
 	c.Assert(err, gc.IsNil)
 
-	_, err = state.GetSecretBackend(context.Background(), secretbackend.BackendIdentifier{ID: backendID1})
-	c.Assert(err, gc.ErrorMatches, `secret backend not found: "`+backendID1+`"`)
-	_, err = state.GetSecretBackend(context.Background(), secretbackend.BackendIdentifier{ID: backendID2})
-	c.Assert(err, gc.ErrorMatches, `secret backend not found: "`+backendID2+`"`)
+	_ = state.RunAtomic(context.Background(), func(ctx domain.AtomicContext) error {
+		_, err := state.GetSecretBackend(ctx, secretbackend.BackendIdentifier{ID: backendID1})
+		c.Assert(err, gc.ErrorMatches, `secret backend not found: "`+backendID1+`"`)
+		_, err = state.GetSecretBackend(ctx, secretbackend.BackendIdentifier{ID: backendID2})
+		c.Assert(err, gc.ErrorMatches, `secret backend not found: "`+backendID2+`"`)
```

Comment:
We're avoiding c.Assert() inside a txn. Return the error and assert outside the RunAtomic.